vscode/src/jdkDownloader/action.ts

```diff
@@ -0,0 +1,299 @@
+import { commands, OpenDialogOptions, OutputChannel, window, workspace } from "vscode";
+import { JdkDownloaderView } from "./view";
+import { OPEN_JDK_VERSION_DOWNLOAD_LINKS, ORACLE_JDK_BASE_DOWNLOAD_URL } from "../constants";
+import * as path from 'path';
+import * as fs from 'fs';
+import { calculateChecksum, downloadFileWithProgressBar, httpsGet } from "../utils";
+import * as cp from 'child_process';
+import { promisify } from "util";
+import { l10n } from "../localiser";
+
+export class JdkDownloaderAction {
+    public static readonly MANUAL_INSTALLATION_TYPE = "manual";
+    public static readonly AUTO_INSTALLATION_TYPE = "automatic";
+    private readonly DOWNLOAD_DIR = path.join(__dirname, 'jdk_downloads');
+
+    private jdkType?: string;
+    private jdkVersion?: string;
+    private osType?: string;
+    private machineArch?: string;
+    private installType?: string;
+    private installationPath?: string | null;
+    private downloadFilePath?: string;
+    private downloadUrl?: string;
+
+    constructor(private readonly logger: OutputChannel, private readonly downloaderView: JdkDownloaderView) { }
+
+    public attachListener = async (message: any) => {
+        const { command, id, jdkVersion, jdkOS, jdkArch, installType } = message;
```

Comment:
can we add parameters as  whatever we need in arguments? instead of entire message as it appears like we dont need entire message to be passed?